### ISSUE TITLE
10 - Neural compressor trained directly on N-body simulations (no pre-training)

### ISSUE THREAD

**Issue by Justinezgh** [Ciela-Institute/CosmOrford#10]
> ⚠️  If you were previously using `configs/experiments/efficientnet_v2_s.yaml`, those results are **invalid** — that config used the wrong model class and loss. Please re-run everything with the corrected config below.

  ## Task

  Train EfficientNetV2-S directly on N-body data using the corrected config `configs/experiments/efficientnet_v2_s_logp.yaml`. Run the full evaluation pipeline (see #7  ) across all budgets.

  ```bash
  trainer fit -c configs/experiments/efficientnet_v2_s_logp.yaml \
      --data.init_args.max_train_samples=<budget>

  python scripts/run_npe_budget_scan.py

**Comment by noe-dia** [comment]
Figure of merit results for this issue. Is this what we would expect ? 

<img width="890" height="551" alt="Image" src="https://github.com/user-attachments/assets/af5da4b3-058a-4aa2-b7f3-d168259c20fa" />



After thinking about it some more, I think this is because we evaluate the log prob on the validation set every 500 optimizer steps that the FOM looks so similar for every budget. I'll try a run where I evaluate the log prob on the validation set more frequently to see if this is the issue.

**Comment by noe-dia** [comment]
After updating the `run_npe_budget_scan.py` script: 

<img width="1200" height="750" alt="Image" src="https://github.com/user-attachments/assets/263e364d-4079-45c0-891d-e2158ea8117a" />
